demisto_sdk/commands/common/hook_validations/mapper.py

```diff
@@ -178,30 +185,26 @@ def is_incident_field_exist(self, id_set_file, is_circle) -> bool:
             click.secho("Skipping mapper incident field validation. Could not read id_set.json.", fg="yellow")
             return True
 
-        built_in_fields = [field.lower() for field in BUILT_IN_FIELDS] + LAYOUT_AND_MAPPER_BUILT_IN_FIELDS
+        content_incident_fields = get_all_incident_and_indicator_fields_from_id_set(id_set_file, 'mapper') + [
+            field.lower() for field in BUILT_IN_FIELDS
+        ] + LAYOUT_AND_MAPPER_BUILT_IN_FIELDS
 
-        content_incident_fields = get_all_incident_and_indicator_fields_from_id_set(id_set_file, 'mapper')
+        invalid_incident_fields = []
+        mapping_type = self.current_file.get('type', {})
 
-        invalid_inc_fields_list = []
         mapper = self.current_file.get('mapping', {})
-        for key, value in mapper.items():
+        for value in mapper.values():
             incident_fields = value.get('internalMapping', {})
-
-            for inc_name, inc_info in incident_fields.items():
-                # for incoming mapper
-                if self.current_file.get('type', {}) == "mapping-incoming":
-                    if inc_name not in content_incident_fields and inc_name.lower() not in built_in_fields:
-                        invalid_inc_fields_list.append(inc_name)
-
-                # for outgoing mapper
-                if self.current_file.get('type', {}) == "mapping-outgoing":
-                    # for inc timer type: "field.StartDate, and for using filters: "simple": "".
-                    if inc_info['simple'] not in content_incident_fields and inc_info['simple'] not in built_in_fields \
-                            and inc_info['simple'].split('.')[0] not in content_incident_fields and inc_info['simple']:
-                        invalid_inc_fields_list.append(inc_name) if inc_info['simple'] else None
-
-        if invalid_inc_fields_list:
-            error_message, error_code = Errors.invalid_incident_field_in_mapper(invalid_inc_fields_list)
+            invalid_incident_fields.extend(
```

Comment:
do we add to this list anywhere else? Why not just initialize here

Reply:
this list should be initialized before the for-loop, otherwise we would overrun its content (if this is what you meant in this case).

its initialized in lines 192